Update device trust docs (#30821)
gravitational/teleport: docs/pages/access-controls/device-trust/enforcing-device-trust.mdx

```code
$ tsh ssh (=clusterDefaults.nodeIP=)
ERROR: ssh: rejected: administratively prohibited (unauthorized device)
```

<Admonition type="tip" title="Trusted Clusters">
It is possible to use [trusted
clusters](../../management/admin/trustedclusters.mdx) to limit the impact of
device mode `required`. A leaf cluster in mode `required` will enforce access to
all of its resources, without imposing the same restrictions to the root
cluster. Likewise, a root cluster will not enforce device trust on resources in
leaf clusters.
</Admonition>

## Locking a device

Similar to [session and identity locking](../guides/locking.mdx), a device can
be locked using `tctl lock`.

Locking blocks certificate issuance and ongoing or future accesses originating
from a locked device. Locking a device only works if device trust is enabled and
if the device is enrolled to Teleport.

Find a device ID to lock:

```code
$ tctl devices ls
Asset Tag     OS    Enroll Status   Device ID
------------ -----  ------------- ------------------------------------
(=devicetrust.asset_tag=)  macOS  enrolled     (=devicetrust.device_id=)
```

Lock a device:

```code

> $ tctl lock --device (=devicetrust.device_id=) --ttl=12h

Reviewer: ```suggestion
$ tctl lock --device=(=devicetrust.device_id=) --ttl=12h
```

description: Learn how to enforce trusted devices with Teleport
videoBanner: gBQyj_X1LVw
---

<Admonition type="warning" title="Preview Note: Supported Resources">
The device trust preview only support SSH, Database and Kubernetes resources.
Support for other resources is planned for upcoming Teleport versions.
</Admonition>

Resources protected by the device mode "required" will enforce the use of a
trusted device, in addition to establishing the user's identity and enforcing
the necessary roles. Furthermore, users using a trusted device leave audit
trails that include the device's information.

Device Trust enforcement can be configured with the following three modes of operation, represented
by the `device_trust_mode` authentication setting:

- `off` - disables device trust. Device authentication is not performed and
  device-aware audit logs are absent.
- `optional` - enables device authentication and device-aware audit, but does
  not require a trusted device to access resources.
- `required` - enables device authentication and device-aware audit.
  Additionally, it requires a trusted device for all SSH, Database and
  Kubernetes connections.

### Prerequisites

> (!docs/pages/includes/commercial-prereqs-tabs.mdx!)

Reviewer 2: added signed `tsh` prereqs here.

## Role-based trusted device enforcement

Role-based configuration enforces trusted device access at the role level. It
can be configured with the `spec.options.device_trust_mode.(required,optional,off)` option and
applies to the resources in its `allow` rules. It
works similar to [`require_session_mfa`](../guides/per-session-mfa.mdx).

To enforce authenticated device checks for a specific role, update the role with
the following:

```diff
kind: role
version: v6
metadata:

>   name: example-role

Reviewer 3: Let's use/reference `require-trusted-device` here. I don't think "example-role" is a good role name for the docs.

by the `device_trust_mode` authentication setting:

- `off` - disables device trust. Device authentication is not performed and
  device-aware audit logs are absent.
- `optional` - enables device authentication and device-aware audit, but does
  not require a trusted device to access resources.
- `required` - enables device authentication and device-aware audit.
  Additionally, it requires a trusted device for all SSH, Database and
  Kubernetes connections.

### Prerequisites
(!docs/pages/includes/commercial-prereqs-tabs.mdx!)

(!docs/pages/includes/device-trust/prereqs.mdx!)

## Role-based trusted device enforcement

Role-based configuration enforces trusted device access at the role level. It
can be configured with the `spec.options.device_trust_mode` option and
applies to the resources in its `allow` rules. It
works similar to [`require_session_mfa`](../guides/per-session-mfa.mdx).

<Admonition type="tip" title="v13.3.6+">
  Teleport version 13.3.6 and above supports preset `require-trusted-device` role to enforce Device Trust.
  You have an option to either update existing role as referenced below, or use the preset role.

>   Ensure to update rules defined in preset `require-trusted-device` role as required.

Reviewer 3: ```suggestion
  Make sure you update the "allow" rules in the role according to your requirements.
```

</Admonition>

To enforce authenticated device checks for a specific role, update the role with
the following:

```diff
kind: role
version: v6
metadata:
  name: example-role
spec:
  options:
    # require authenticated device check for this role
+   device_trust_mode: "required" # add this line
  allow:
    logins: ['admin']
    kubernetes_groups: ['edit']
    node_labels:
      '*': '*'
    ...

```

```code
$ tctl create -f device-enforcement.yaml
```

```code

> $ tctl create -f admin.yaml

Reviewer 3: This seems out of place, what is "admin.yaml" ?

```

> 

Reviewer 3: nit:

```suggestion
```

## Cluster-wide trusted device enforcement

Cluster-wide configuration enforces trusted device access at the cluster level.

<Admonition type="note" title="Default configuration">
Enterprise clusters run in `optional` mode by default. Changing the mode to
`required` will enforce a trusted device for all SSH, Database and Kubernetes
accesses.
</Admonition>

> 

Reviewer 3: nit:

```suggestion
```

> <Admonition type="warning" title="Web UI">

Reviewer 3: Double-admonition.

Reviewer 2: The web ui is important one but I think the admonition block above this "Enterprise clusters run in `optional` mode by default. Changing the mode to `required` will enforce a trusted device for all SSH, Database and Kubernetes
accesses." can be added as a normal sentence.

```

You can also edit this configuration directly:

```code
$ tctl edit cluster_auth_preference
```

</TabItem>
<TabItem label="Static Config" options="Self-Hosted">
Edit the Auth Server's `teleport.yaml` file and restart all Auth Services:

```diff
auth_service:
  authentication:
    type: local
    second_factor: "on"
    webauthn:
      rp_id: teleport.example.com
    device_trust:
+     mode: "required" # add this line
```

</TabItem>
</Tabs>

Once the config is updated, SSH, Database and Kubernetes access without a trusted device will be forbidden.
For example, SSH access without a trusted device fails with the following error:

```code
$ tsh ssh server1
ERROR: ssh: rejected: administratively prohibited (unauthorized device)
```

<Admonition type="tip" title="Trusted Clusters">
It is possible to use [trusted
clusters](../../management/admin/trustedclusters.mdx) to limit the impact of
device mode `required`. A leaf cluster in mode `required` will enforce access to
all of its resources, without imposing the same restrictions to the root
cluster. Likewise, a root cluster will not enforce device trust on resources in
leaf clusters.
</Admonition>

> <Admonition type="tip" title="Per-role vs. Cluster-wide">

Reviewer 3: Double-admonition.

> Per-role configuration let's you progressively block unauthorized device access,

Reviewer 3: ```suggestion
Per-role configuration lets you progressively block unauthorized device access,
```

and is useful for getting started with Teleport Device Trust.

Once all the devices used are enrolled in Teleport, cluster-wide

> configuration will ensure every resources can only ever be accessed with a

Reviewer 3: ```suggestion
enforcement will ensure every resources can only ever be accessed with a
```

> trusted device.

Reviewer 2: I am removing that entire comparison admonition block.

</Admonition>

> ## Locking a device

Reviewer 2: I think it belongs in enforcement but tagging @zmb3 for third opinion.
